hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MulticastJoiner.java

```diff
@@ -148,7 +141,7 @@ public void onMessage(Object msg) {
                 logger.warning(e);
             }
         } finally {
-            node.multicastService.removeMulticastListener(listener);
+            splitBrainJoinMessages.clear();
```

Comment:
I think expiration is not required. If an old split brain join message is processed, then either the clusters should still merge (win), or if some other changes have occurred in the meanwhile (eg already merged so sending address is not still the master of other cluster, cluster version changed to incompatible version on one cluster) then there will be no merge.